### PR TITLE
Update Low Precision LayerNorm arguments

### PR DESCRIPTION
**Pull request by nik-mosaic**
This PR changes the Low Precision LayerNorm (LPLN) arguments to match the torch.nn.LayerNorm arguments, so a user can manually replace a LayerNorm layer with an LPLN layer. Please note that using
`apply_low_precision_layernorm(model)`
will replace all torch.nn.LayerNorms for you, so it is not necessary to use the LPLayerNorm class directly.

On exporting models with LPLN:
LowPrecision LayerNorm works with ONNX exporting, but it does not support Torchscript. torch.get_autocast_gpu_dtype() is not supported by Torchscript.

We change the test `test_gpu_huggingface_export_for_inference_onnx` to use LPLN instead of Fused LayerNorm to verify export functionality.

### CODE REVIEW

File: composer/algorithms/low_precision_layernorm/low_precision_layernorm.py [hunk 1]
```diff
@@ -106,27 +104,38 @@ def forward(self, x):
...
+def _cast_if_autocast_enabled(tensor):
+    if torch.is_autocast_enabled():
+        if tensor.device.type == 'cuda':
+            dtype = torch.get_autocast_gpu_dtype()
+        elif tensor.device.type == 'cpu':
+            dtype = torch.get_autocast_cpu_dtype()
+        else:
+            raise NotImplementedError()
+        return tensor.to(dtype=dtype)
+    return tensor
```
Comment (mvpatel2000) [inline composer/algorithms/low_precision_layernorm/low_precision_layernorm.py:116]: Why this change? If its necessary can we add to LPGN

File: composer/algorithms/low_precision_layernorm/low_precision_layernorm.py [hunk 2]
```diff
@@ -106,27 +104,38 @@ def forward(self, x):
...
         )
 
 
 def _to_LPLayerNorm(layer: torch.nn.Module, module_index: int) -> LPLayerNorm:
-    if not isinstance(layer, torch.nn.LayerNorm):
-        raise TypeError(f'Expected torch.nn.LayerNorm, got {type(layer)}')
-    return LPLayerNorm(layer)
+    """Defines a replacement policy from a `torch.nn.LayerNorm` to a `LPLayerNorm`"""
+    assert isinstance(layer,
+                      torch.nn.LayerNorm), 'The replacement policy will look for all instances of torch.nn.LayerNorm'
```
Comment (mvpatel2000) [inline composer/algorithms/low_precision_layernorm/low_precision_layernorm.py:129]:
```suggestion
    if not isinstance(layer, torch.nn.LayerNorm):
        raise TypeError(f'Expected torch.nn.LayerNorm, got {type(layer)}')
```

**Review by mvpatel2000 — APPROVED** [review]

File: composer/algorithms/low_precision_layernorm/low_precision_layernorm.py [hunk 3]
```diff
@@ -106,27 +104,38 @@ def forward(self, x):
...
+    lp_layernorm = LPLayerNorm(layer.normalized_shape, layer.eps, layer.elementwise_affine, layer.weight.device,
+                               layer.weight.dtype)
+    with torch.no_grad():
+        lp_layernorm.weight.copy_(layer.weight)
+        lp_layernorm.bias.copy_(layer.bias)
+    return lp_layernorm
 
 
-def to_FusedLayerNorm(layer: torch.nn.Module, module_index: int) -> APEXFusedLayerNorm:
+def _to_FusedLayerNorm(layer: torch.nn.Module, module_index: int) -> APEXFusedLayerNorm:
```
Comment (dskhudia) [inline composer/algorithms/low_precision_layernorm/low_precision_layernorm.py:138]: Change this name to _to_LowPrecisionLayerNorm also similar to how you have changed fused => low precision.

File: tests/utils/test_inference.py [hunk 4]
```diff
@@ -163,7 +162,7 @@ def test_gpu_huggingface_export_for_inference_onnx():
     import onnxruntime as ort
     import transformers
 
-    from composer.functional import apply_fused_layernorm
```
Comment (dskhudia) [inline tests/utils/test_inference.py:166]: How did this work? There was no apply_fused_layernorm function?

**Review by dskhudia — APPROVED** [review]

File: tests/utils/test_inference.py (hunk 4 above)
Comment (mvpatel2000) [reply tests/utils/test_inference.py:166]: It is in the fused_layernorm.py algorithm
